### PR TITLE
Handle subclasses for Kafka Serde autodetection

### LINKED ISSUE #42353: Kafka deserializer autodetection is not working for classes, implementing the Deserializer interface indirectly
### Describe the bug

Brief: Auto-detection seems to work, when a concrete deserializer directly implements `org.apache.kafka.common.serialization.Deserializer`, but fails to be looked up, when this interface is being indirectly (e.g. via a base class) inherited.

### Expected behavior

Auto-detection works for all classes, implementing Kafka's Deserializer interface.

### Actual behavior

Auto-detection fails to discover this potential deserializer class.

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

Linux JIRA-ITC-1752 5.15.137.3-microsoft-standard-WSL2 #1 SMP Mon Nov 6 23:32:38 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "17.0.12" 2024-07-16

### Quarkus version or git rev

3.12.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.6 (bc0240f3c744dd6b6ec2920b3cd08dcc295161ae)

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ozangunalp**
- Fixes #42353

### CODE REVIEW

File: extensions/kafka-client/runtime/pom.xml [hunk 1]
```diff
@@ -88,6 +88,18 @@
...
+                <artifactId>jandex-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>make-index</id>
+                        <goals>
+                            <goal>jandex</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
```
Comment (geoand) [inline extensions/kafka-client/runtime/pom.xml:102]: We usually don't do this, but instead add the empty `beans.xml` file.

Comment (ozangunalp) [reply extensions/kafka-client/runtime/pom.xml:102]: For runtime modules of extensions, I've mostly seen the plugin usage.

Comment (geoand) [reply extensions/kafka-client/runtime/pom.xml:102]: It works, but it's not the preferred way, so let's use the `beans.xml` way for new stuff please.

Thanks

Comment (ozangunalp) [reply extensions/kafka-client/runtime/pom.xml]: Finally, I went with adding individual classes to the index.

Comment (geoand) [reply extensions/kafka-client/runtime/pom.xml]: Even better 😉

**Review by geoand — APPROVED** [review]

**Pull request merged**